File: app/src/main/java/org/doomer/qnotez/viewmodel/NoteAddViewModel.java
package org.doomer.qnotez.viewmodel;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.os.AsyncTask;

import org.doomer.qnotez.db.AppDatabase;
import org.doomer.qnotez.db.NoteModel;
import org.doomer.qnotez.utils.NoteUtils;

public class NoteAddViewModel extends AndroidViewModel {
    protected AppDatabase database;

    public NoteAddViewModel(Application application) {
        super(application);

        database = AppDatabase.getDatabase(this.getApplication());
    }

    public void addNote(final NoteModel item) {
        new NoteUtils.NoteAddAsyncTask(database).execute(item);
    }
}

File: app/src/main/kotlin/org/doomer/qnotez/viewmodel/NoteAddViewModel.kt
package org.doomer.qnotez.viewmodel

import android.arch.lifecycle.ViewModel

import javax.inject.Inject

import org.doomer.qnotez.db.AppDatabase
import org.doomer.qnotez.db.NoteModel
import org.doomer.qnotez.utils.NoteUtils


class NoteAddViewModel @Inject constructor(private val db : AppDatabase) : ViewModel() {

    fun addNote(item: NoteModel) {
        NoteUtils.NoteAddAsyncTask(db).execute(item)
    }
}


//class NoteAddViewModel(application: Application) : AndroidViewModel(application) {
//    protected var database: AppDatabase
//
//    init {
//
//        database = AppDatabase.getDatabase(this.getApplication())
//    }
//
//    fun addNote(item: NoteModel) {
//        NoteUtils.NoteAddAsyncTask(database).execute(item)
//    }
//}
